test/unit/alignment/multiple/align_multiple_seqan2_adaptation_test.cpp

```diff
@@ -0,0 +1,86 @@
+#include <seqan3/alignment/configuration/align_config_scoring_scheme.hpp>
+#include <seqan3/alignment/multiple/detail/align_multiple_seqan2_adaptation.hpp>
+#include <seqan3/alignment/scoring/aminoacid_scoring_scheme.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/core/detail/test_accessor.hpp>
+
+namespace seqan3::detail
+{
+struct test_accessor
+{
+    template <typename alphabet_type, typename config_t>
+    static auto initialise_scoring_scheme(config_t const & config)
+    {
+        seqan3::detail::align_multiple_seqan2_adaptation<alphabet_type> seqan2_adaptation;
+        return seqan2_adaptation.initialise_scoring_scheme(config);
+    }
+};
+} // seqan3::detail
+
+template <typename alphabet_type>
+using adaptation_t = seqan3::detail::align_multiple_seqan2_adaptation<alphabet_type>;
+
+TEST(basics, alphabet_conversion)
+{
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::dna4>::alphabet_type, seqan::Dna>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::dna5>::alphabet_type, seqan::Dna5>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::dna15>::alphabet_type, seqan::Iupac>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::rna4>::alphabet_type, seqan::Rna>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::rna5>::alphabet_type, seqan::Rna5>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::aa27>::alphabet_type, seqan::AminoAcid>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::aa10li>::alphabet_type, seqan::ReducedAminoAcid<seqan::Li10>>));
+    EXPECT_TRUE((std::same_as<adaptation_t<seqan3::aa10murphy>::alphabet_type,
+                              seqan::ReducedAminoAcid<seqan::Murphy10>>));
+}
+
+TEST(initialise_scoring_scheme, config_no_scoring_configuration)
+{
+    // no scoring information
+    seqan3::configuration config = seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-2},
+                                                                             seqan3::gap_open_score{-8}}};
+
+    auto msaOpt = seqan3::detail::test_accessor::initialise_scoring_scheme<seqan3::dna4>(config);
+
+    EXPECT_TRUE((std::same_as<decltype(msaOpt.sc), seqan::Score<int>>));
+}
+
+TEST(initialise_scoring_scheme, blosum62)
+{
+    // no scoring information
+    seqan3::aminoacid_scoring_scheme scheme{seqan3::aminoacid_similarity_matrix::BLOSUM62};
+    seqan3::configuration config = seqan3::align_cfg::scoring_scheme{scheme};
+
+    auto msaOpt = seqan3::detail::test_accessor::initialise_scoring_scheme<decltype(scheme)::alphabet_type>(config);
+
+    // compare matrix size and abort test if not equal so the value comparison does not segfault
+    ASSERT_EQ(static_cast<size_t>(decltype(msaOpt.sc)::TAB_SIZE), static_cast<size_t>(seqan::Blosum62::TAB_SIZE));
+
+    seqan::Blosum62 expected_matrix{};
+    for (size_t i = 0; i < static_cast<size_t>(seqan::Blosum62::TAB_SIZE); ++i)
+        EXPECT_EQ(msaOpt.sc.data_tab[i], expected_matrix.data_tab[i]);
```

Comment:
well, the check should not be on the internals but on the public interfaces.

Reply:
I don't know if I get you right here, why should we not test the detail interfaces?
Note that the public interface also tests the BLOSUM matrix for the correct MSA (see align_multiple_test).